src/Strategy/RPM.hs

```diff
@@ -0,0 +1,132 @@
+discover = walk $ \dir _ files ->
+  case find (\f -> ".spec" `isSuffixOf` fileName f) files of
+    Nothing -> pure WalkContinue
+    Just specFile -> do
+      runSimpleStrategy "rpm-spec" RustGroup $ fmap (mkProjectClosure dir) (analyze specFile)
+      pure WalkSkipAll
+
+analyze :: (Has ReadFS sig m, Has (Error ReadFSErr) sig m) => Path Rel File -> m (Graphing Dependency)
+analyze specFile = do
+  specFileText <- readContentsText specFile
+  pure . buildGraph $ getSpecDeps specFileText
+
+mkProjectClosure :: Path Rel Dir ->  Graphing Dependency -> ProjectClosureBody
+mkProjectClosure dir graph = ProjectClosureBody
+  { bodyModuleDir    = dir
+  , bodyDependencies = dependencies
+  , bodyLicenses     = []
+  }
+  where
+  dependencies = ProjectDependencies
+    { dependenciesGraph    = graph
+    , dependenciesOptimal  = NotOptimal
+    , dependenciesComplete = NotComplete
+    }
+
+toDependency :: RPMDependency -> Set SpecFileLabel -> Dependency
+toDependency pkg = foldr applyLabel Dependency 
+  { dependencyType = CargoType 
+  , dependencyName = rpmDepName pkg
+  , dependencyVersion = rpmConstraint pkg
+  , dependencyLocations = []
+  , dependencyEnvironments = []
+  , dependencyTags = M.empty
+  }
+  where
+    applyLabel :: SpecFileLabel -> Dependency -> Dependency
+    applyLabel (RequiresType env) dep = dep { dependencyEnvironments = env : dependencyEnvironments dep }
+
+buildGraph :: Dependencies -> Graphing Dependency
+buildGraph Dependencies{..} = run . withLabeling toDependency $ do
+  traverse_ direct depBuildRequires
+  traverse (flip label $ RequiresType EnvDevelopment) depBuildRequires
+
+-- buildConstraint tail = RPMConstraint version <$> comparator
+buildConstraint :: Text -> Maybe VerConstraint
+buildConstraint tail = constraint
+  where
+    (comparatorStr, rawVersion) = splitOnceOn " " $ T.strip tail
+    version = T.strip rawVersion
+    constraint = case T.strip comparatorStr of
+      "<=" -> Just $ CLessOrEq version
+      "<" -> Just $ CLess version
+      ">=" -> Just $ CGreaterOrEq version
+      ">" -> Just $ CGreater version
+      "=" -> Just $ CEq version
+      "==" -> Just $ CEq version
+      _ -> Nothing
+
+splitOnceOn :: Text -> Text -> (Text, Text)
+splitOnceOn needle haystack = (head, strippedTail)
+  where
+    len = T.length needle
+    (head, tail) = T.breakOn needle haystack
+    strippedTail = T.drop len tail
+
+getTypeFromLine :: Text -> Maybe RequiresType
+getTypeFromLine line = safeReq
+  where
+    (header, value) = splitOnceOn ": " line
+    (pkgName, rawConstraint) = splitOnceOn " " $ T.strip value
+    isSafeName name = not $ "%{" `T.isInfixOf` name
+    safeReq = if isSafeName pkgName then req else Nothing
+    req = case header of
+      "BuildRequires" -> Just . BuildRequires . RPMDependency pkgName $ buildConstraint rawConstraint
+      _ -> Nothing
+
+buildDependencies :: [RequiresType] -> Dependencies
+buildDependencies = foldr addDep blankDeps
```

Comment:
[nit] I find functions like this easier to read if they are not written in point free. Not a really strong opinion here though. Another thing for us to discuss in a style guide.

Reply:
I think with complex type sigs or tons of internal logic, it can be pretty tough to read.  When you have a single monadic parameter, I think it reduces noise slightly, especially with collections like `List`.  The type sig tells you what you need, and the one-liner is the logic.  Without the where clause, point-free would be super confusing, I'd agree then.

Reply:
hlint would warn here if the variable was present -- this is a pretty typical form found elsewhere in the codebase, e.g., in `toDependency` above.

Something that would help here, though, are type signatures on at least `addDep`

Reply:
hlint would warn here if the variable was present -- this is a pretty typical form found elsewhere in the codebase, e.g., in `toDependency` above.

Something that would help here, though, are type signatures on at least `addDep`

Reply:
Good counter example to point-free is on the `isSafeName` in the `where` clause of `getTypeFromLine`.  Point-free would be more confusing there, even though it's easily done.